polaris/ocean/convergence/spherical/analysis.py

```diff
@@ -0,0 +1,403 @@
+        dependencies = self.dependencies_dict
+
+        for resolution in self.resolutions:
+            mesh_name = resolution_to_subdir(resolution)
+            base_mesh = dependencies['mesh'][resolution]
+            init = dependencies['init'][resolution]
+            forward = dependencies['forward'][resolution]
+            self.add_input_file(
+                filename=f'{mesh_name}_mesh.nc',
+                work_dir_target=f'{base_mesh.path}/base_mesh.nc')
+            self.add_input_file(
+                filename=f'{mesh_name}_init.nc',
+                work_dir_target=f'{init.path}/initial_state.nc')
+            self.add_input_file(
+                filename=f'{mesh_name}_output.nc',
+                work_dir_target=f'{forward.path}/output.nc')
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        plt.switch_backend('Agg')
+        convergence_vars = self.convergence_vars
+        for var in convergence_vars:
+            self.plot_convergence(
+                variable_name=var["name"],
+                title=var["title"],
+                zidx=var["zidx"])
+
+    def plot_convergence(self, variable_name, title, zidx):
+        """
+        Compute the error norm for each resolution and produce a convergence
+        plot
+
+        Parameters
+        ----------
+        variable_name : str
+            The name of the variable as given in the output netcdf file
+
+        title : str
+            The name of the variable to use in the plot title
+
+        zidx : int
+            The z-index to use for variables that have an nVertLevels
+            dimension, which should be None for variables that don't
+        """
+        resolutions = self.resolutions
+        logger = self.logger
+        conv_thresh, error_type = self.convergence_parameters(
+            field_name=variable_name)
+
+        error = []
+        for resolution in resolutions:
+            mesh_name = resolution_to_subdir(resolution)
+            error_res = self.compute_error(
+                mesh_name=mesh_name,
+                variable_name=variable_name,
+                zidx=zidx,
+                error_type=error_type)
+            error.append(error_res)
+
+        res_array = np.array(resolutions)
+        error_array = np.array(error)
+        filename = f'convergence_{variable_name}.csv'
+        data = np.stack((res_array, error_array), axis=1)
+        df = pd.DataFrame(data, columns=['resolution', error_type])
+        df.to_csv(f'convergence_{variable_name}.csv', index=False)
+
+        convergence_failed = False
+        poly = np.polyfit(np.log10(res_array), np.log10(error_array), 1)
+        convergence = poly[0]
+        conv_round = np.round(convergence, 3)
+
+        fit = res_array ** poly[0] * 10 ** poly[1]
+
+        order1 = 0.5 * error_array[-1] * (res_array / res_array[-1])
+        order2 = 0.5 * error_array[-1] * (res_array / res_array[-1])**2
+
+        fig = plt.figure()
```

Comment:
We might want to use the `matplotlib` style file that we use in the global viz plots.

We would need these imports (from `polaris.viz.globe`)
```
import sys
from typing import TYPE_CHECKING  # noqa: F401

if TYPE_CHECKING or sys.version_info >= (3, 9, 0):
    import importlib.resources as imp_res  # noqa: F401
else:
    # python <= 3.8
    import importlib_resources as imp_res  # noqa: F401
...
```
and then:
```suggestion
    style_filename = str(
        imp_res.files('polaris.viz') / 'polaris.mplstyle')
    plt.style.use(style_filename)

    fig = plt.figure()
```
Without this, the plots have different font sizes before and after the viz has run:
![convergence_icos](https://github.com/E3SM-Project/polaris/assets/4179064/45622bd1-27df-4e78-8254-10fc8e91bd1f)
![convergence_qu](https://github.com/E3SM-Project/polaris/assets/4179064/2d5fb893-e07f-4b2c-99df-06edd27544a7)

Reply:
Let's decide if we want to do this in a separate PR...